assets/javascripts/discourse/services/chat-emoji-picker-manager.js

```diff
@@ -0,0 +1,146 @@
+import { headerOffset } from "discourse/lib/offset-calculator";
+import { createPopper } from "@popperjs/core";
+import Service from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { bind } from "discourse-common/utils/decorators";
+import { later, schedule } from "@ember/runloop";
+import { makeArray } from "discourse-common/lib/helpers";
+import { Promise } from "rsvp";
+import { computed } from "@ember/object";
+
+const TRANSITION_TIME = 125; // CSS transition time
+
+export default class ChatEmojiPickerManager extends Service {
+  @tracked opened = false;
+  @tracked closing = false;
+  @tracked loading = false;
+  @tracked context = null;
+  @tracked emojis = null;
+  @tracked visibleSections = ["favorites", "smileys_&_emotion"];
+  @tracked lastVisibleSection = "favorites";
+  @tracked element = null;
+  @tracked callback;
+
+  @computed("emojis.[]", "loading")
+  get sections() {
+    return !this.loading && this.emojis ? Object.keys(this.emojis) : [];
+  }
+
+  @bind
+  closeExisting() {
+    this.callback = null;
+    this.opened = false;
+    this.visibleSections = ["favorites", "smileys_&_emotion"];
+    this.lastVisibleSection = "favorites";
+  }
+
+  @bind
+  close() {
+    this.callback = null;
+    this.closing = true;
+
+    later(() => {
+      if (this.isDestroyed || this.isDestroying) {
+        return;
+      }
+
+      this.visibleSections = ["favorites", "smileys_&_emotion"];
+      this.lastVisibleSection = "favorites";
```

Comment:
Looks like these two are defaults and also set on `closeExisting`. Should we move them to a constant?